src/rpc/scgi_task.cc

```diff
@@ -239,36 +250,102 @@ SCgiTask::event_error() {
   close();
 }
 
+// On failure, returns false and the buffer is left in an
+// indeterminate state, but m_position and m_bufferSize remain the
+// same.
+bool
+SCgiTask::gzip_compress_response(const char* buffer, uint32_t length, std::string_view header_template) {
```

Comment:
This is a bit too ugly a solution, not what I had in mind.

Add to utils a function like this:

```
gzip_compress(const char* buffer, uint32_t length, std::function<bool(char*,uint32_t,uint32_t)>)
```

The lambda function will write from m_buffer+max_header_size (where max_header_size is calculated from the current header template plus max length of the printed response size).

It will receive deflateBound value so it can resize on the first call.

Once done writing, copy the bytes from the header and response size to the buffer so they end at the start of the written gzip'ed data.

Don't use snprintf for a whole template, instead the header should be two static const strings, and you copy them and the response size.